utils_real_test.go

```diff
@@ -76,21 +77,44 @@ func installDistro(t *testing.T, ctx context.Context, distroName string, rootfs
 }
 
 // uninstallDistro checks if a distro exists and if it does, it unregisters it.
-func uninstallDistro(distro wsl.Distro) error {
+func uninstallDistro(distro wsl.Distro, allowShutdown bool) error {
 	if r, err := distro.IsRegistered(); err == nil && !r {
 		return nil
 	}
-	cmd := fmt.Sprintf("$env:WSL_UTF8=1 ; wsl.exe --unregister %s", distro.Name())
-	_, err := exec.Command("powershell.exe", "-command", cmd).CombinedOutput() //nolint:gosec
+
+	// 1. Attempt unregistering
+	unregisterCmd := fmt.Sprintf("$env:WSL_UTF8=1 ; wsl.exe --unregister %q", distro.Name())
+	err := exec.Command("powershell.exe", "-Command", unregisterCmd).Run() //nolint:gosec
+	if err == nil {
+		return nil
+	}
+
+	// 2. Attempt terminate, then unregister
+	cmd := fmt.Sprintf("$env:WSL_UTF8=1 ; wsl.exe --terminate %q", distro.Name())
+	_ = exec.Command("powershell.exe", "-Command", cmd).Run()                      //nolint:gosec
```

Comment:
Hum, ignoring the error here? What happens if this is errors out and the nex command is ran?

Reply:
Ohh I guess this is a good opportunity for Go 1.20 joined errors!